### ISSUE TITLE
Add env name to the "aws-exports.js"

### ISSUE THREAD

**Issue by corelike** [aws-amplify/amplify-cli#4364]
**Is your feature request related to a problem? Please describe.**
I need to access the enviornment name of my backend in the frontend to display different features.

**Describe the solution you'd like**
Add the value "env" in the aws-exports.js which is generated by the Amplify CLI

**Describe alternatives you've considered**
Query the backend stack of amplify or replace a config file during the build process

**Comment by hisham** [comment]
+1. But there's an easier workaround available if you have a REST or Graphql endpoint in the aws-exports. These endpoints are always post-fixed with the backend env. So you can do something like this: 

```
import awsmobile from '../../../shared/aws-exports.js';

const restEndpoint = awsmobile.aws_cloud_logic_custom[0].endpoint;
const backendEnv = restEndpoint.split('amazonaws.com/')[1];
```

**Comment by dylan-westbury** [comment]
Yes this would be great, currently we need to do a custom build through webpack to set the env. 

This is fine for bigger projects, but for smaller projects it would be much more simple to access via aws-exports.

**Comment by malcomm** [comment]
Any updates on this? This seems pretty simple to implement and would be very useful. The workaround above does not work for me (I have graphql and my URL doesn't have anything on it that would indicated the environment).

**Comment by ktoyoda39** [comment]
I also would like to have this environment name in "aws-exports.js" .
The workaround above doesn't work for my environment too.

**Comment by charlieforward9** [comment]
```
// uses the bucket name to determine the environment
function detectEnvironment(config): 'prod' | 'staging' | 'dev' {
  const bucketName = config.aws_user_files_s3_bucket;
  const parts = bucketName.split('-');
  const env = parts[parts.length - 1];

  if (env === 'prod' || env === 'staging' || env === 'dev') {
    return env;
  } else {
    throw new Error(`Unknown environment: ${env}`);
  }
}
```

Should work for me, having 3 environments (`prod`, `staging` and `dev`). It pulls the suffix of the bucket name, which matches my environment. The only issue I am having right now is the fact that I want it to be in Typescript, and the config does not have a typedef defined within amplify. 

Ideally, this 5 year old can be closed with a simple new field soon though..
